### ISSUE TITLE
Windows installation

### ISSUE THREAD

**Issue by ChrisRackauckas** [chakravala/Reduce.jl#11]
```julia
julia> Pkg.build("Reduce")
INFO: Building Reduce
Downloading reduce binaries...
Installing reduce binaries...
DONE

julia> Pkg.test("Reduce")
INFO: Testing Reduce
getheap: HeapAlloc failed
Reduce: A Portable General-Purpose Computer Algebra SystemError During Test
  Test threw an exception of type Base.UVError
  Expression: rcall($(Expr(:quote, :((1 + pi)^ 2)))) == convert(Expr, RExpr(rcall("(1+pi)**2")))
  write: broken pipe (EPIPE)
  Stacktrace:
   [1] try_yieldto(::Base.##296#297{Task}, ::Task) at .\event.jl:189
   [2] wait() at .\event.jl:234
   [3] uv_write(::Base.PipeEndpoint, ::Ptr{UInt8}, ::UInt64) at .\stream.jl:811
   [4] unsafe_write(::Base.PipeEndpoint, ::Ptr{UInt8}, ::UInt64) at .\stream.jl:832
   [5] clear at C:\Users\Chris\.julia\v0.6\Reduce\src\Reduce.jl:80 [inlined]
   [6] write(::Reduce.PSL, ::String) at C:\Users\Chris\.julia\v0.6\Reduce\src\Reduce.jl:87
   [7] #rcall#33(::Array{Symbol,1}, ::Array{Symbol,1}, ::Function, ::Reduce.RExpr) at C:\Users\Chris
\.julia\v0.6\Reduce\src\rexpr.jl:384
   [8] (::Reduce.#kw##rcall)(::Array{Any,1}, ::Reduce.#rcall, ::Reduce.RExpr) at .\<missing>:0
   [9] #rcall#34(::Array{Symbol,1}, ::Array{Symbol,1}, ::Function, ::Expr) at C:\Users\Chris\.julia\
v0.6\Reduce\src\rexpr.jl:422
   [10] rcall(::Expr) at C:\Users\Chris\.julia\v0.6\Reduce\src\rexpr.jl:422
   [11] include_from_node1(::String) at .\loading.jl:576
   [12] include(::String) at .\sysimg.jl:14
   [13] process_options(::Base.JLOptions) at .\client.jl:305
   [14] _start() at .\client.jl:371
ERROR: LoadError: There was an error during testing
while loading C:\Users\Chris\.julia\v0.6\Reduce\test\runtests.jl, in expression starting online 6
=========================================[ ERROR: Reduce ]==========================================

failed process: Process(`'C:\Users\Chris\AppData\Local\Julia-0.6.2\bin\julia.exe' -Cx86-64 '-JC:\Use
rs\Chris\AppData\Local\Julia-0.6.2\lib\julia\sys.dll' --compile=yes --depwarn=yes --check-bounds=yes
 --code-coverage=none --color=yes --compilecache=yes 'C:\Users\Chris\.julia\v0.6\Reduce\test\runtest
s.jl'`, ProcessExited(1)) [1]

====================================================================================================
ERROR: Reduce had test errors
```

this is after adding `64BIT` to `/deps/osbit.txt`

**Comment by chakravala** [comment]
Alright, so I have made a new commit on `master` with the `osbit.txt` issue resolved, tested Reduce.jl on Windows 7 (32 bit) and it seems to be working fine for me.

The reason you might be getting a heap allocation error might be due to the differences between 32bit and 64 bit memory allocations.

> In PSL the binary executable bpsl knows an argument -td [mega]bytes. Suitable values for this are coded in the calling scripts, e.g. redpsl, and need not be given by the user. When referring to MB in connection with -td, we make the convention that 1 MB = 10⁶ B in contrast to 1024² B.
>
> On 32 bit systems one common default choice for -td is 16 MB. That is, after startup there remain 16 MB – 8 MB for BPS – size of bpsl ≤ 8 MB ≈ 2·10⁶ cells. On 32 bit systems, where the heap cannot be enlarged, the -td is ignored and there are is the maximum of 128 MB allocated, which amounts to approximately 32·10⁶ cells.
> 
> On 64 bit systems, if the argument of -td is less than 10⁶, then is multiplied by 10⁶ thus describing MB in the sense above. The default choice is 1000 (MB). There are generally 2 heaps used so that after startup the available heap amounts to (1000 MB – 8 MB for BPS – size of bpsl) ÷ 2 ≤ 496 MB ≈ 62·10⁶ cells. 

Could you maybe experiment with this value to see if that's the issue?

https://github.com/chakravala/Reduce.jl/blob/68fee6ca074582df34f34397e9d43bb1d009fbc4/src/Reduce.jl#L51

This might also be related to why the `appveyor` build didn't work. Please let me know if you experimented with the `-td` value or if you continue to be unsuccessful at making it work on 64bit.

**Comment by ChrisRackauckas** [comment]
what's `dirf` supposed to be? Give it as a runnable code and I'll run it.

**Comment by chakravala** [comment]
Don't worry about `dirf`, it's the directory for the binary. Just edit the numerical value in the src/Reduce.jl  after `-td` and see if a different value for it will make it work.

**Comment by chakravala** [comment]
Alternatively, the solution to this issue would be to make CSL version of Reduce the default instead of PSL. I just tested it, and the CSL version should also work fine on Windows and does not require the memory size to be specified in advance. This was an intended change I had planned anyway.

Ideally, it would be good to know what value to use for `-td` for the PSL version on Windows, but I think we can live with the CSL version as default, since it also supports unicode characters (unlike PSL). However, the PSL version is known to be about twice as fast, but since in Julia the main time consumption goes into the `Pipe`, it actually makes no difference which version is used in terms of speed at the moment.

I will make a new commit which uses CSL by default soon. This will not require you to `build` the package again, it's only a change to the package source.

However, please do experiment with the `-td` value to see if you can find a working number.

**Comment by chakravala** [comment]
In conclusion, I have committed the necessary changes to the build script and to the package source to change the default Reduce binary for Windows users from PSL to CSL. Additionally, I was able to make the distribution of the binary work with the `appveyor` automated testing.

However, there is one quirk for Windows users that I am not able to understand yet. When Windows users use the `@force using Reduce.Algebra` command, they are not able to import `*` for some reason. This can be circumvented by manually setting `* = Algebra.:*` so it's no big issue, but strange.

Still need to implement the change from PSL to CSL for Linux and OSX users for consistency.

This issue is now resolved.
